### PR TITLE
Feature/KX-59: migrate to Typegoose

### PR DESCRIPTION
**Pull request by dlste**

### CODE REVIEW

File: client/src/components/form/Form.tsx [hunk 1]
```diff
@@ -20,8 +20,10 @@ import { useState, useEffect } from "react";
 import locales from "./../../locales.json";
 import { PreviewMap, SatMap } from "../map/Map";
 import API from "../../API";
-import { AreaType, DocCoords, KxDocumentType, Scale, Stakeholders } from "../../enum";
+import { AreaType, KxDocumentType, Scale, Stakeholders } from "../../enum";
+import { DocCoords } from "../../model";
```
Comment (GiuseppeArbore) [inline client/src/components/form/Form.tsx:24]: you can import DocCoords and KxDocuments together

File: server/src/db/dao.ts [hunk 2]
```diff
@@ -1,6 +1,8 @@
 import { KxDocument } from "../models/model";
-import { Document } from "./schema";
-import mongoose from 'mongoose';
+//import { Document } from "./schema";
+import { KxDocumentModel } from "../models/model";
+//import mongoose from 'mongoose';
```
Comment (GiuseppeArbore) [inline server/src/db/dao.ts:4]: remove if useless

File: server/src/db/dao.ts [hunk 3]
```diff
@@ -1,6 +1,8 @@
 import { KxDocument } from "../models/model";
-import { Document } from "./schema";
-import mongoose from 'mongoose';
+//import { Document } from "./schema";
```
Comment (GiuseppeArbore) [inline server/src/db/dao.ts:2]: remove if useless

File: server/src/routes.ts [hunk 4]
```diff
@@ -18,23 +18,23 @@ export function initRoutes(app: Application) {
             .withMessage('Invalid stakeholder value'),
         body('scale').notEmpty().withMessage('Scale is required')
             .isNumeric().withMessage('Scale must be a number'),
-        body('scale_info').notEmpty().withMessage('Scale info is required')
-            .isIn(Object.values(Scale)).withMessage('Invalid scale value'),
+        //body('scale_info').notEmpty().withMessage('Scale info is required')
+        //    .isIn(Object.values(Scale)).withMessage('Invalid scale value'),
```
Comment (GiuseppeArbore) [inline server/src/routes.ts:22]: remove if useless

**Review by lbordino — APPROVED** [review]
Are we passing now Object Id or string when doing the connections?

File: client/src/model.ts [hunk 5]
```diff
@@ -1,64 +1,110 @@
...
+
+    // TODO: use proper schema and validate
+    @prop({type: [mongoose.Schema.Types.Mixed], validate: () => {}})
+    pages?: PageRange[];
+
+    @prop({required: true, type: Connections})
+    connections!: Connections;
 }
 
+//export const KxDocumentModel = getModelForClass(KxDocument);
```
Comment (GiuseppeArbore) [inline client/src/model.ts:110]: remove if useless

**Review by GiuseppeArbore — APPROVED** [review]

**Pull request merged**